code/game/machinery/computer/tram_controls.dm

```diff
@@ -15,14 +15,26 @@
 	//find the tram, late so the tram is all... set up so when this is called? i'm seriously stupid and 90% of what i do consists of barely educated guessing :)
 	find_tram()
 
+/**
+ * Finds the tram from the console
+ *
+ * Locates tram parts in the lift global list after everything is done.
+ */
 /obj/machinery/computer/tram_controls/proc/find_tram()
-	var/obj/structure/industrial_lift/tram/tram_struct = locate(/obj/structure/industrial_lift/tram) in GLOB.lifts
-	tram_part = tram_struct //possibly setting to something null, that's fine, but
+	var/obj/structure/industrial_lift/tram/central/tram_loc = locate() in GLOB.lifts
+	tram_part = tram_loc //possibly setting to something null, that's fine, but
 	tram_part.find_our_location()
 
 /obj/machinery/computer/tram_controls/ui_state(mob/user)
 	return GLOB.not_incapacitated_state
 
+/obj/machinery/computer/tram_controls/ui_status(mob/user,/datum/tgui/ui)
+	if(tram_part.travelling)
+		return UI_CLOSE
+	if(!in_range(user, src) && !isobserver(user))
+		return UI_CLOSE
+	return ..()
```

Comment:
What's your goal here? Cause I'm pretty sure there's a better way to do it

Reply:
The cause of the blender was the fact the UI wouldn't update properly if left open and you walked away, meaning you could tell the tram to go to a destination it was already at and brrrrrrrrrr.

TGUI really isn't my strong suite and forcing it closed guarantees the info on the UI is updated so no blender.

Reply:
aight